src/Forms/Form.php

```diff
@@ -1823,4 +1808,34 @@ protected function buildRequestHandler()
     {
         return FormRequestHandler::create($this);
     }
+
+    /**
+     * Can the body of this form be cached?
+     *
+     * Figure out if we can cache this form
+     * - forms with validation shouldn't be cached, because their error messages won't be shown
+     * - forms with security tokens shouldn't be cached because security tokens expire
+     *
+     * @return bool
+     */
+    public function canBeCached()
+    {
+        if ($this->getSecurityToken()->isEnabled()) {
+            return false;
+        }
+        if ($this->FormMethod() !== 'GET') {
```

Comment:
This conditional is a bit weird (forms can display sensitive or session-specific data regardless of HTTP verb), but it's retained for backwards compat. It's probably a good idea not to mention it in the PHPDoc to avoid devs relying on this behaviour (it's not mentioned at the moment, and I don't think we need to fix that)

Reply:
Ok removing the phpdoc for now.

Reply:
Well, the URL changes if the form is submitted by GET and so HTTP Caching is mess relevant, right?

If someone goes to `http://example.com/home/Form?PersonalInfo=Dave`, they'll get that personal info regardless of whether the CDN has a cache or not...

The problem before was that errors / personal info was shown in the "vanilla" url `http://example.com/home/Form` because the submission was POSTed... right?

Reply:
I guess it's flawed logic if the data gets stored in the session and the user redirected back, though...